include/iceflow/iceflow.hpp

```diff
@@ -95,10 +95,8 @@ class IceFlow {
 
     m_running = true;
     while (m_running) {
-      std::chrono::time_point<std::chrono::steady_clock>
+      std::optional<std::chrono::time_point<std::chrono::steady_clock>>
```

Comment:
We could replace this with `std::chono::milliseconds::max()` instead of adding an `std::optional`

Assuming that at least one producer exists (which should be ensured after line 108), its duration until next publish will always be smaller than `std::chrono::milliseconds::max()`.

However, one thing i just noticed is that we don't ensure that the list of producers is not modified during a loop iteration, which might cause some issues regardless of whether we use `std::optional` or `std::chrono::milliseconds::max()`.
Maybe we could extend the `m_producerRegistrationMutex` lock to be kept for the entire time of a loop iteration (and the entire methods for adding/removing producers, respectively)?

Reply:
Hmm, this commit should already be enough to cover the second part, right? 72c2aa0ab5fbb69a03623c47ac31b9c3b6872f42

Reply:
Hmm, there is one small bug that has been introduced by https://github.com/hsel-netsys/iceflow/commit/fa81bea86d7bc884e56279f4b62e5b426a53c149 now – I will fix that and then proceed with merging, thank you for the approval, @pulsastrix!

Reply:
The issue should now be fixed by 980cf2086830feac76e035a7b2ea427255e0967d Maybe you can have a final look here, then I would squash and merge :)